lib/utils/houchdb.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const PouchW = require('pouchdb-wrapper');
+const _ = require('lodash');
+const defaultOptions = {
+    namespace: 'default',
+    port: 5984,
+    hostname: 'localhost',
+    protocol: 'http',
+    db: 'my-db',
+    plugins: []
+};
+
+const register = function(server, options, next) {
+    options = _.merge(defaultOptions, options);
+
+    // register PouchDB plugins
+    // this appears safe to do repeatedly, every time we call this function
+    _.forEach(options.plugins, (plugin) => {
+        PouchW.PouchDB.plugin(plugin);
+    });
+
+    let config;
+    if (options.path) {
+        config = {
+            path: options.path,
+            name: options.name
+        };
+    } else if (options.conn) {
+        config = {
+            name: options.name,
+            conn: options.conn
+        };
+    }
+
+    const db = new PouchW(config);
+    const pluginName = module.exports.attributes.name;
+    const exposableObj = {};
+    exposableObj[options.name] = db;
+    exposableObj.constructor = PouchW;
+
+    // throw error if namespace already exists
+    if (
+        server.plugins[pluginName] &&
+        server.plugins[pluginName][options.name]
+    ) {
+        const msg = 'PouchDB namespace `' + options.prefix + '` already exists';
+        next(new Error(msg));
+    }
+
+    // attempt to get info from the db
+    return db.info()
```

Comment:
warning, if the pouch instance is pointing a _remote_ couch instance and the couch store has not yet been generated, this will fail!  we ran into this yesterday  on COINSTAC server where purging out all dbs then running the tests _fails_.  Once content is added or the db has been created remotely, then tests pass.  otherwise creating a new `PouchDB` or `PouchW` instance and doing an `.info` returns a 404